icu4j/main/tests/core/src/com/ibm/icu/dev/test/number/DecimalQuantityTest.java

```diff
@@ -603,6 +605,127 @@ public void testNickelRounding() {
         }
     }
 
+    @Test
+    public void testCompactDecimalPluralOperandHelperMethods() {
+        ULocale locale = new ULocale("fr-FR");
+        LocalizedNumberFormatter noCompactSimple = NumberFormatter.with().notation(Notation.simple()).locale(locale);
+        LocalizedNumberFormatter compactLong = NumberFormatter.with().notation(Notation.compactLong()).locale(locale);
+        LocalizedNumberFormatter compactShort = NumberFormatter.with().notation(Notation.compactShort()).locale(locale);
+
+        // 123456789
+        DecimalQuantity_DualStorageBCD dq1NoCompact = (DecimalQuantity_DualStorageBCD) noCompactSimple.format(123456789).getFixedDecimal();
+        DecimalQuantity_DualStorageBCD dq1CompactLong = (DecimalQuantity_DualStorageBCD) compactLong.format(123456789).getFixedDecimal();
+        DecimalQuantity_DualStorageBCD dq1CompactShort = (DecimalQuantity_DualStorageBCD) compactShort.format(123456789).getFixedDecimal();
+        assertEquals("compact decimal toLong", 123456789, dq1NoCompact.toLong(false));
```

Comment:
I have a preference for these tests to be written in the parameterized style.  It helps highlight what is changing from line to line.  It would also increase readability if you could assertEquals for the string in addition to the decimal quantity.  For example, the parameters could be,

```java
Object[][] cases = new Object[][] {
    // formatter, input, string output, long output, double output, suppressed exponent
    { compactShort, 123456789, "123M", 123000000, 123000000.0, 6 }
    // ...
}
```

It makes it more readable and also easier to add more cases.

Reply:
Done.